cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -1154,7 +1154,7 @@ class ObjectInputFile final : public io::RandomAccessFile {
 // AWS doc says "5 MB" but it's not clear whether those are MB or MiB,
 // so I chose the safer value.
 // (see https://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadUploadPart.html)
-static constexpr int64_t kMinimumPartUpload = 5 * 1024 * 1024;
+static constexpr int64_t kMinimumPartUpload = 10 * 1024 * 1024;
```

Comment:
Previously our limit was ~2.4TB. In the new design, it is 100GB. It would be only 50GB if it was left at 5MB per part. I feel like this is an okay balance, but willing to hear other's opinions.

Reply:
It seems ok to me, it just means that we're using a bit more RAM to buffer outgoing bytes.

Reply:
Can you add a comment about the limit, though?